### PR TITLE
Replace explicit 'delete' calls with std::unique_ptr in AST

### PR DESCRIPTION
**Pull request by jrose-apple**
No functionality change.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Review by ahoppen — APPROVED** [review]
LGTM

File: include/swift/AST/Module.h [hunk 1]
```diff
@@ -1164,7 +1162,7 @@ class SourceFile final : public FileUnit {
   /// If not None, the underlying vector should contain tokens of this source file.
   Optional<std::vector<Token>> AllCorrectedTokens;
 
-  SourceFileSyntaxInfo &SyntaxInfo;
+  std::unique_ptr<SourceFileSyntaxInfo> SyntaxInfo;
```
Comment (jrose-apple) [inline include/swift/AST/Module.h]: Wait, this doesn't need to be out-of-line at all.

**Review by nkcsgexi — APPROVED** [review]

File: include/swift/AST/Module.h (hunk 1 above)
Comment (jrose-apple) [reply include/swift/AST/Module.h]: …yes it does, never mind.

**Pull request merged**